lib/util/http/cache/abstract-http-cache-provider.ts

```diff
@@ -43,11 +46,28 @@ export abstract class AbstractHttpCacheProvider implements HttpCacheProvider {
     return Promise.resolve(null);
   }
 
+  private preventCaching<T>(resp: HttpResponse<T>): boolean {
+    if (this.checkCacheControlHeader === false) {
+      return false;
+    }
+
+    const isPublic = resp.headers?.['cache-control']
+      ?.toLocaleLowerCase()
+      .split(regEx(/\s*,\s*/))
+      .includes('public');
+
+    return !isPublic;
```

Comment:
I think we should allow caching if the header is missing 🤔

we should first add some logging and track some stats in the hosted app. so we're sure most ecosystems behave properly. and if not we need config options to workaround.

Reply:
The only place it's being used is NPM. And it's likely to be so for long time, because usually we tend to use cache decorators.